crud.py

```diff
@@ -125,34 +131,30 @@ async def delete_charge(charge_id: str) -> None:
 async def check_address_balance(charge_id: str) -> Optional[Charges]:
     charge = await get_charge(charge_id)
     assert charge
-    if not charge.paid:
-        if charge.onchainaddress:
-            try:
-                balance = await fetch_onchain_balance(charge)
-                confirmed = int(balance["confirmed"])
-                unconfirmed = int(balance["unconfirmed"])
-                if confirmed != charge.balance or unconfirmed != charge.pending:
-                    if charge.zeroconf:
-                        confirmed += unconfirmed
-
-                    await update_charge(
-                        charge_id=charge_id,
-                        balance=confirmed,
-                        pending=unconfirmed,
-                    )
-            except Exception as e:
-                logger.warning(e)
-        if charge.lnbitswallet:
-            try:
-                invoice_status = await api_payment(charge.payment_hash)
-
-                if invoice_status["paid"]:
-                    return await update_charge(
-                        charge_id=charge_id, balance=charge.amount
-                    )
-            except Exception as e:
-                logger.warning(e)
-    return await get_charge(charge_id)
+    if charge.paid:
+        return await get_charge(charge_id)
+    if charge.onchainaddress:
+        try:
+            balance = await fetch_onchain_balance(charge)
+            confirmed = int(balance["confirmed"])
+            unconfirmed = int(balance["unconfirmed"])
+            if confirmed != charge.balance or unconfirmed != charge.pending:
+                if charge.zeroconf:
+                    confirmed += unconfirmed
+
+                return await update_charge(
+                    charge_id=charge_id,
+                    balance=confirmed,
+                    pending=unconfirmed,
+                )
+        except Exception as e:
+            logger.warning(e)
+    if charge.lnbitswallet and charge.payment_hash:
+        payment = await get_standalone_payment(charge.payment_hash)
+        status = (await payment.check_status()).success if payment else False
+        if status:
+            return await update_charge(charge_id=charge_id, balance=charge.amount)
+    return None
```

Comment:
the old code was always returning an instance no matter of the if-else branches and of the errors:
```py
return await get_charge(charge_id)
```

Reply:
done